src/core/dense_set.cc

```diff
@@ -180,21 +180,34 @@ void* DenseSet::PopDataFront(DenseSet::ChainVectorIterator it) {
 }
 
 void DenseSet::ClearInternal() {
+  constexpr unsigned kArrLen = 32;
+  ClearItem arr[kArrLen];
+  unsigned len = 0;
+
   for (auto it = entries_.begin(); it != entries_.end(); ++it) {
-    while (!it->IsEmpty()) {
-      bool has_ttl = it->HasTtl();
-      bool is_displ = it->IsDisplaced();
-      void* obj = PopDataFront(it);
-      int32_t delta = int32_t(BucketId(obj, 0)) - int32_t(it - entries_.begin());
-      if (is_displ) {
-        DCHECK(delta < 2 || delta > -2);
-      } else {
-        DCHECK_EQ(delta, 0);
-      }
-      ObjDelete(obj, has_ttl);
+    if (it->IsEmpty())
+      continue;
+
+    DensePtr ptr = *it;
+    auto& dest = arr[len++];
+    dest.has_ttl = ptr.HasTtl();
+
+    PREFETCH_READ(ptr.Raw());
+    if (ptr.IsObject()) {
+      dest.obj = ptr.Raw();
+      dest.ptr.Reset();
+    } else {
+      dest.ptr = ptr;
+      dest.obj = nullptr;
+    }
+    if (len == kArrLen) {
+      ClearBatch(kArrLen, arr);
+      len = 0;
     }
```

Comment:
Nice batching here :) Just curious why `constexpr unsigned kArrLen = 32;` ? Have you played around and increased this batch limt ?

Reply:
i did yeah. it can probably be differrent for differrent CPUs but it more or less near a sweet point